packages/eas-cli/src/build/validate.ts

```diff
@@ -0,0 +1,60 @@
+import { Workflow } from '@expo/eas-build-job';
+import fs from 'fs-extra';
+import path from 'path';
+
+import Log, { learnMore } from '../log';
+import vcs from '../vcs';
+import { BuildContext } from './context';
+import { Platform } from './types';
+
+export function checkNodeEnvVariable(ctx: BuildContext<Platform>) {
+  if (ctx.buildProfile.env?.NODE_ENV === 'production') {
+    Log.warn(
+      'You are setting environment variable NODE_ENV=production, remember that it will be available during entire build process and will affect among other things what yarn/npm packages will be installed.'
```

Comment:
```suggestion
      'You set NODE_ENV=production in the build profile. Remember that it will be available during the entire build process. In particular, it will make yarn/npm install only production packages.'
```